### PR TITLE
Create 0000-allow-github-release-url-shorthand.md

### PR DESCRIPTION
**Pull request by jabrown85**
[Readable](https://github.com/buildpack/rfcs/blob/d750008a099ea3f8a3367fabbadd904cea42b8c4/text/0000-allow-github-release-url-shorthand.md)

### CODE REVIEW

**Comment by hone** [comment]
The convenience of ubiquity of GitHub support in bundler for Ruby was a well liked feature. GitHub is getting a package registry feature giving repos a docker registry.  should we be using that with buildpackages / distribution here?

With respect to vendor neutrality, I wonder if we need to add support for other vendors too and where does that stop? Are `pack` plugins eventually a solution that that problem?

**Comment by nebhale** [comment]
I think I concur with @hone that the forthcoming build packages, which are image-based, likely mean that the URIs will start pointing at Docker Registry coordinates.  I'm not sure how this relates to the URIs one would then use though.

File: text/0000-allow-github-release-url-shorthand.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+```
+pack build myimage --buildpack https://github.com/company/my-buildpack@v1.0.0
+pack build myimage --buildpack https://github.com/company/my-buildpack
+```
+
+The proposal would have pack use with the [GitHub Releases API](https://developer.github.com/v3/repos/releases/) when a url matches the above example patterns. A GitHub release would be found using the tag name specified in the url. If no tag is specified, the latest release is used.
+
+If no matching release is found, an error will be generated stating the buildpack could not be found. 
+
+If a matching release is found, the assets of that release will be searched for an entry with a `content_type` of `application/gzip`. This found asset's `browser_download_url` will become the target for the buildpack download process. If no matching asset is found, GitHub's default tar would become the target. This would fallback would allow repositories that are also a valid CNB to release without uploading an explicit tar.
```
Comment (ekcasey) [inline text/0000-allow-github-release-url-shorthand.md]: I'm also not sure it is safe to assume that a buildpack release should have exactly one gzip artifact. I can imagine buildpack authors releasing two buildpack artifacts that are built from the same source but are configured/compiled/packaged to support different stacks.

**Comment by jabrown85** [comment]
Closing this. We think the distribution and registry will be OK for now. We might revisit at a later time if the need comes back up.

**Pull request closed without merging**